src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1080,34 +1080,17 @@ private static TextEdit BuildRevertedEdit(TextEdit provisionalTextEdit)
             return response?.Response;
         }
 
-        public async override Task<VSInternalHover?> HoverAsync(DelegatedPositionParams request, CancellationToken cancellationToken)
```

Comment:
The diff on this looks super weird, but basically the three methods (so far) that just deal with TextDocumentPositionParams all do exactly the same thing, the only thing different is the LSP method they call, so I abstract a helper method.

I'm not sure if its obvious enough why these three are special, and can share, but rename (for example) can't, so let me know if this feels weird and I can revert. Of course, we could make the helper method take a params object, and then rename can call it too, but I'll leave that for a future PR. If there aren't more than one endpoint that would need to pass in params, I'd probably perfer to keep it separate.

Reply:
I'm in support of the helper method since it makes things cleaner, although admittedly it would be nice to have rename use the helper method too (although that can be in a future PR like you mentioned).